### PR TITLE
Added a parser for daily and modified the appearance of the change co…

### PR DESCRIPTION
**Pull request by GithinjiHans**
…lor button

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
The test doesn't yet take/return a goal object as described in the issue.

**Review by tijlleenders — APPROVED** [review]
Think @Tushar-4781 should review too before we merge

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**
